rest/import_test.go

```diff
@@ -779,3 +1009,53 @@ func assertDocProperty(t *testing.T, getDocResponse *TestResponse, propertyName
 	assertTrue(t, ok, fmt.Sprintf("Expected property %s not found in response %s", propertyName, getDocResponse.Body.Bytes()))
 	assert.Equals(t, value, expectedPropertyValue)
 }
+
+func rawDocWithSyncMeta() string {
+
+	return `
+{
+    "_sync": {
+        "rev": "1-ca9ad22802b66f662ff171f226211d5c",
+        "sequence": 1,
+        "recent_sequences": [
+            1
+        ],
+        "history": {
+            "revs": [
+                "1-ca9ad22802b66f662ff171f226211d5c"
+            ],
+            "parents": [
+                -1
+            ],
+            "channels": [
+                null
+            ]
+        },
+        "cas": "",
+        "time_saved": "2017-11-29T12:46:13.456631-08:00"
+    }
+}
+`
+
+}
+
+func assertXattrSyncMetaRevGeneration(t *testing.T, bucket base.Bucket, key string, expectedRevGeneration int) {
+	xattr := map[string]interface{}{}
+	_, err := bucket.GetWithXattr(key, "_sync", nil, &xattr)
+	assertNoError(t, err, "Error Getting Xattr")
+	revision, ok := xattr["rev"]
+	assert.True(t, ok)
+	generation, _ := db.ParseRevID(revision.(string))
+	log.Printf("assertXattrSyncMetaRevGeneration generation: %d rev: %s", generation, revision)
+	assert.True(t, generation == expectedRevGeneration)
+}
+
+func assertXattrSyncMeta(t *testing.T, bucket base.Bucket, key string) {
```

Comment:
There's benefit in reusing MetaRevGeneration here, to catch cases where we're doing unwanted multiple updates.